src/nvim/api/private/helpers.c

```diff
@@ -760,12 +761,7 @@ bool object_to_vim(Object obj, typval_T *tv, Error *err)
     case kObjectTypeWindow:
     case kObjectTypeTabpage:
     case kObjectTypeInteger:
-      if (obj.data.integer > VARNUMBER_MAX
-          || obj.data.integer < VARNUMBER_MIN) {
-        api_set_error(err, kErrorTypeValidation, "Integer value outside range");
-        return false;
-      }
-
+      STATIC_ASSERT(sizeof(obj.data.integer) <= sizeof(varnumber_T), "Expected integer size to be less than or equal to VimL number size");
```

Comment:
Looks like too long line. Below there also is `--n;` which is not going to pass the linter.